sky/server/stream_utils.py

```diff
@@ -86,10 +87,12 @@ async def log_streamer(request_id: Optional[str],
                 # Use smaller padding (1024 bytes) to force browser rendering
                 yield f'{waiting_msg}' + ' ' * 4096 + '\n'
             # Sleep shortly to avoid storming the DB and CPU and allow other
-            # coroutines to run. This busy waiting loop is performance critical
-            # for short-running requests, so we do not want to yield too long.
-            await asyncio.sleep(0.1)
-            request_task = requests_lib.get_request(request_id)
+            # coroutines to run.
+            # TODO(aylei): we should use a better mechanism to avoid busy
+            # polling the DB, which can be a bottleneck for high-concurrency
+            # requests.
+            await asyncio.sleep(0.5)
+            request_task = await requests_lib.get_request_async(request_id)
```

Comment:
Another optimization can be done: we can only query the status of the request instead of the entire request body, which might be smaller and faster to load.